src/scripts/main.js

```diff
@@ -1,3 +1,80 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+
+table.addEventListener('click', (e) => {
+  const bodies = document.querySelector('tbody');
+  const bodyRows = [...bodies.rows];
+
+  const data = bodyRows.map((el) => {
+    const newObject = {
+      name: el.cells[0].textContent,
+      position: el.cells[1].textContent,
+      age: el.cells[2].textContent,
+      salary: el.cells[3].textContent,
+    };
+
+    return newObject;
+  });
+
+  bodies.remove();
+
+  const newBodies = document.createElement('tbody');
+
+  table.tHead.after(newBodies);
+
+  let sortedData = [];
+
+  switch (e.target.textContent) {
+    case 'Name': {
+      sortedData = [...data].sort((el1, el2) => {
```

Comment:
Consider creating variable with `[...data]` and avoid to make a copy action in every case